xsuite/src/cli/verifyReproducibleCmd.ts

```diff
@@ -0,0 +1,159 @@
+import { Command } from "commander";
+import { globSync } from "glob";
+import { cwd, log } from "../context";
+import { Proxy } from "../proxy/proxy";
+import { pause, logError } from "./helpers";
+
+const defaultVerifierUrl = "https://devnet-play-api.multiversx.com";
+const pollDelay = process.env.VITEST_WORKER_ID ? 1 : 5000;
+
+export const addVerifyReproducibleCmd = (cmd: Command) => {
+  cmd
+    .command("verify-reproducible")
+    .description("Verify a contract built in a reproducible way.")
+    .argument(
+      "[DIR]",
+      "Directory in which the command is executed (default: $(PWD))",
+    )
+    .requiredOption("--sc <SC>", "Address of the smart contract to be verified")
+    .option(
+      "--verifier-url <VERIFIIER_URL>",
+      `Verifier URL (default: ${defaultVerifierUrl})`,
+    )
+    .action(publishReproducible);
+};
+
+export const publishReproducible = async (
+  dirArgument: string | undefined,
+  {
+    sc,
+    verifierUrl,
+  }: {
+    sc: string;
+    verifierUrl?: string;
+  },
+) => {
+  let dir: string;
+  if (dirArgument !== undefined) {
+    dir = path.resolve(dirArgument);
+  } else {
+    dir = cwd();
+  }
+  verifierUrl = verifierUrl ?? defaultVerifierUrl;
+
+  log("Verification started...");
+  const foundSourceFiles = globSync(`${dir}/**/*.source.json`);
+  if (foundSourceFiles.length == 0) {
+    logError(
+      `Cannot find any source file to verify in ${dir}, which is required to verify a smart contract.`,
+    );
+    return;
+  }
+  if (foundSourceFiles.length !== 1) {
+    logError(
+      `Found more than one source file to verify in ${dir}. Source files found:`,
+    );
+    foundSourceFiles.forEach((f) => logError(f));
+    return;
+  }
+
+  log(`Found smart contract at ${path.dirname(foundSourceFiles[0])}`);
+  const sourceCode = JSON.parse(fs.readFileSync(foundSourceFiles[0], "utf-8"));
+  const image = sourceCode.metadata.buildMetadata.builderName;
+  if (!image) {
+    logError(`Could not read image name from ${foundSourceFiles[0]}.`);
+    return;
+  }
+
+  log(`Smart contract was built with image: ${image}.`);
+  const request = new ContractVerificationRequest(
+    sc,
+    sourceCode,
+    "",
+    image,
+  ).toDictionary();
+  await verifyAndWait(verifierUrl, request);
+};
+
+const verifyAndWait = async (baseUrl: string, request: any) => {
+  const startTime = new Date().getTime();
+  const proxy = new Proxy({
+    proxyUrl: baseUrl,
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  log("Requesting verification...");
+  const response = await proxy.fetchRaw("/verifier", request);
+
+  const taskId = response.taskId;
+  if (!taskId) {
+    throw Error(`Verification failed. Response: ${JSON.stringify(response)}`);
+  }
+
+  log(`Verification in process (taskId: ${taskId})...`);
+  log("Please wait while we verify your contract. This may take a while.");
+
+  let oldStatus = "";
+  let status = "";
+
+  while (status != "finished") {
```

Comment:
Also, forgot to mention that comparisons should be "===" and "!==" (instead of "==" and "!=") whenever possible